### DISCUSSION TITLE
log messages to text widget?

### DISCUSSION THREAD

**Discussion by unrushed-boots** [slint-ui/slint#4270] [Q&A]
with a text widget in a scrollable is there an example for printing log messages? either `log` or `tracing` is fine. seems like using in-out variable and newlines is ok, but cant pass the widget to the trait to write to. all my testing/demo code prints to stdout not the gui. there also seems an async issue with `tracing` when trying. not sure what to do from here with it. any help or possibly new example in repo?

**Comment by ogoffart** [comment]
you will need to do something like this:  (i have not tested to run this code, just wrote it on the spot)

```
let message = ... the message to log ...;
slint::invoke_from_event_loop(move || {
   LOG_WINDOW.with(|log_window| { // the log_window is stored in a thread_local
        log_window.invoke_add_log(message); // invoke a function `add-log` declared in your .slint file
   });
});
```
